Live-Snap/Live-Snap/Controllers/ExportWallpaperViewController.swift

```diff
@@ -193,7 +200,158 @@ class ExportWallpaperViewController: UIViewController, PHLivePhotoViewDelegate {
         }
     }
     
+}
+
+fileprivate class SaveSuccessPopUpView: UIView {
+    
+    var successIconImageView: UIImageView!
+    var successLabel: UILabel!
+    var lineBreakOneView: UIView!
+    var goToPhotosLabel: UILabel!
+    var goToPhotosImageView: UIImageView!
+    var goToPhotosIconButton: UIButton!
+    var lineBreakTwoView: UIView!
+    var lineBreakThreeView: UIView!
+    var newLabel: UILabel!
+    var newImageView: UIImageView!
+    var newIconButton: UIButton!
+    var rateLabel: UILabel!
+    var rateImageView: UIImageView!
+    var rateIconButton: UIButton!
     
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setLayout()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setLayout() {
+        backgroundColor = UIColor.snapBlack
+        layer.cornerRadius = 5
+        layer.masksToBounds = true
+        
+        successIconImageView = UIImageView(frame: CGRect(x: 0.0, y: 0.0, width: 40, height: 40))
+        successIconImageView.center.x = center.x
+        successIconImageView.frame.origin.y = frame.origin.y + 10.0
+        let successIconImage = UIImage(named: "success")?.withRenderingMode(.alwaysTemplate)
+        successIconImageView.image = successIconImage
+        successIconImageView.tintColor = UIColor.snapWhite
+        successIconImageView.contentMode = .scaleAspectFill
+        
+        successLabel = UILabel()
+        successLabel.text = "Success"
+        successLabel.font = successLabel.font.withSize(15)
+        successLabel.sizeToFit()
+        successLabel.center.x = center.x
+        successLabel.frame.origin.y = successIconImageView.frame.maxY + 5.0
+        successLabel.textColor = UIColor.snapWhite
+        
+        lineBreakOneView = UIView(frame: CGRect(x: 0.0, y: 0.0, width: frame.width, height: 1.0))
+        lineBreakOneView.center.x = center.x
+        lineBreakOneView.frame.origin.y = successLabel.frame.maxY + 15.0
+        lineBreakOneView.backgroundColor = UIColor.black
+        
+        goToPhotosLabel = UILabel()
+        goToPhotosLabel.text = "Go to Photos"
+        goToPhotosLabel.font = goToPhotosLabel.font.withSize(15)
+        goToPhotosLabel.sizeToFit()
+        goToPhotosLabel.textColor = UIColor.snapYellow
+        
+        goToPhotosImageView = UIImageView(frame: CGRect(x: 0, y: 0, width: goToPhotosLabel.frame.height, height: goToPhotosLabel.frame.height))
+        let openWallpaperSettingsImage = UIImage(named: "open")?.withRenderingMode(.alwaysTemplate)
+        goToPhotosImageView.image = openWallpaperSettingsImage
+        goToPhotosImageView.tintColor = UIColor.snapYellow
+        goToPhotosImageView.contentMode = .scaleAspectFill
+        
+        goToPhotosIconButton = IconButton(icon: goToPhotosImageView, label: goToPhotosLabel, spaceBetween: 10.0)
+        goToPhotosIconButton.center.x = center.x
```

Comment:
Let me know if there is a better way to implement the logic for the IconButton. Right now, the frame is calculated by the size of the icon, label, and space between both.

Reply:
So the button should not take views as parameters, this makes it unclear what owns the views and could cause problems if we have logic to modify the views in multiple places. Instead you should pass an image, text, and possibly set some options on the button such as space between or alignment (center, left, right), etc